docs/code-docs/source/zero3.rst

```diff
@@ -293,3 +293,37 @@ parallelism to fit them in limited GPU memory.
 
 .. autoclass:: deepspeed.zero.TiledLinear
     :members:
+
+
+Debugging
+---------
+
+Debugging ZeRO training is complicated by the partitioning of parameters, gradients, and optimizer states.
+To simplify debugging, DeepSpeed provides the following routines for accessing individual model states
+in their full form (i.e., un-partitioned).
+
+.. autofunction:: deepspeed.utils.safe_get_full_fp32_param
+
+.. autofunction:: deepspeed.utils.safe_get_full_grad
+
+.. autofunction:: deepspeed.utils.safe_get_full_optimizer_state
+
+
+These routines can be used in a training loop as shown in the following snippet.
```

Comment:
ah, you prefer "routines" instead of "utilities" - that works, let's then rename these back int he paras above, to keep things consistent.